sql/core/src/test/scala/org/apache/spark/sql/FileBasedDataSourceSuite.scala

```diff
@@ -842,6 +842,162 @@ class FileBasedDataSourceSuite extends QueryTest
       }
     }
   }
+
+  test("SPARK-31116: Select simple parquet columns correctly in case insensitive manner") {
+    withSQLConf(SQLConf.CASE_SENSITIVE.key -> "false") {
```

Comment:
Can you explicitly test both cases of tuning `spark.sql.optimizer.nestedSchemaPruning.enabled` off and on too?